themes/classic/templates/catalog/_partials/product-variants.tpl

```diff
@@ -35,4 +35,5 @@
       {/if}
     </div>
   {/foreach}
+  <input type="hidden" name="idCombination" id="idCombination" value="{$product.id_product_attribute}">
```

Comment:
I think it's confusing because you introduced an `input` field.

What about adding it, in the section here: https://github.com/PrestaShop/PrestaShop/blob/develop/themes/classic/templates/catalog/product.tpl#L26

You can add a few attributes like `data-id-product`, `data-id-product-attribute` and such. Anything that seems useful to you.
At least it's clear that it's for JS, input give the impression that you'll need it on the server side.

What do you think?